src/primitives/transaction.cpp

```diff
@@ -43,6 +43,40 @@ std::string CTxIn::ToString() const
     return str;
 }
 
+
+bool CTxIn::SpendsNestedPayToWitnessPubKeyHashOutput(CScript spentScriptPubKey) const {
+    if ((!spentScriptPubKey.IsPayToScriptHash()) || scriptWitness.IsNull()) {
+        return false;
+    }
+
+    return scriptSig.IsNestedPayToWitnessPubKeyHashScriptSig();
+}
+
+bool CTxIn::SpendsNestedPayToWitnessScriptHashOutput(CScript spentScriptPubKey) const {
+    if (!(spentScriptPubKey.IsPayToScriptHash()) || scriptWitness.IsNull()) {
```

Comment:
This looks like it may be problematic; should the `!` be inside the parens? Otherwise it looks like null scriptWitnesses don't cause an immediate return.

Reply:
Ah wait, nevermind - misbalanced parens when reading :)